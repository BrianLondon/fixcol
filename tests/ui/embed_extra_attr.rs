use fixed_derive::{ReadFixed, WriteFixed};

#[derive(ReadFixed)]
struct Point {
    #[fixed(width = 5)]
    point_x: u16,
    #[fixed(width = 5)]
    point_y: u16,
}

#[derive(ReadFixed, WriteFixed)]
#[fixed(key_width = 1)]
enum Alg {
    #[fixed(key = "N")]
    Num(#[fixed(width = 5)] u16),
    #[fixed(key = "P", embed = true)]
    Point(#[fixed(width = 5)] Point),
    #[fixed(key = "Z")]
    Zero,
}

pub fn main() {}
