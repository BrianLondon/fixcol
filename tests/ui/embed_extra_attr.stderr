error: fixcol-derive error: Did not expect fixed attribute on embedded enum variant
  --> tests/ui/embed_extra_attr.rs:17:13
   |
17 |     Point(#[fixed(width = 5)] Point),
   |             ^^^^^
